code/components/gta-streaming-five/src/GameEventHooks.cpp

```diff
@@ -62,18 +107,144 @@ void* HandleEventWrap(void* group, rage::fwEvent* event)
 		}
 		catch (std::exception& e)
 		{
-		
 		}
 	}
 
 	return (*TFunc)(group, event);
 }
 
-#include <sstream>
+template<decltype(g_eventCall1)* TFunc>
+void* HandleEventWrapReact(rage::fwEventGroup* group, rage::fwEvent* event)
+{
+	if (event)
+	{
+		/*
+		 * event:
+		 * These events are stored with different member variables and with different layouts per class/struct,
+		 * so we have to investigate them case by case.
+		 * 
+		 * group:
+		 * In this reacting version the entity that triggers this event is stored in group->entities[17]
+		 * as well as in group->entities[18].
+		 */
 
-namespace rage
+		try
+		{
+			msgpack::sbuffer buf;
+			msgpack::packer packer(buf);
+
+			const char* eventName = typeid(*event).name() + 6;
+
+			packer.pack_array(4); // we'll offer 4 parameters
+			msgpack_pack(packer, eventName, event->GetId(), GetGuidFromBaseSafe(group->entities[17]));
+
+			// Quick 4 bytes check, i.e.: no strcmp
```

Comment:
Why the odd code to optimize out `strcmp`/`memcmp` in a function that is filled with a lot of other definitely slow code paths (like, potentially, typeid, or msgpack packing)?

Reply:
To limit any more performance impact, this way the compiler can optimize the switch cases, while also taking in account there will be more cases added later. Looking at the amount of events triggered at times I wasn't that happy to use strcmp.

It was already doing msgpack serialization, but now it's moved to here. typeid was already used for getting the name.